eskip/template.go

```diff
@@ -39,15 +41,40 @@ func NewTemplate(template string) *Template {
 // Apply evaluates the template using a TemplateGetter function to resolve the
 // placeholders.
 func (t *Template) Apply(get TemplateGetter) string {
-	result := t.template
-
 	if get == nil {
-		return result
+		return t.template
 	}
+	result, _ := t.apply(get)
+	return result
+}
 
+// ApplyContext evaluates the template using a filter context to resolve the
+// placeholders. Returns true if all placeholders resolved to non-empty values.
+func (t *Template) ApplyContext(ctx filters.FilterContext) (string, bool) {
```

Comment:
With a future extension in mind (e.g. support `${request.header.X-Foo}` placeholder) I think here we should have a pair `ApplyRequestContext` and `ApplyResponseContext` to be used in filter's `Request` and `Response` methods respectively.
`ApplyRequestContext` would have access to context and request properties and `ApplyResponseContext` would have access to context, request and response properties.

Reply:
Done